### PR TITLE
Add foreign key support (Phase 1)

### PR DESCRIPTION
**Pull request by seapagan**
## Summary

- Implement Phase 1 of foreign key support with explicit `_id` field access
- Add `ForeignKey()` factory function for defining FK relationships
- Enable SQLite FK constraint enforcement via PRAGMA
- Support all ON DELETE/ON UPDATE actions (CASCADE, SET NULL, RESTRICT, NO ACTION)
- Add comprehensive test suite (28 tests)

## Features

- `ForeignKey()` factory storing metadata in `json_schema_extra`
- `ForeignKeyInfo` dataclass for relationship metadata
- FK exception classes for constraint violation handling
- Auto-indexing of FK columns for query performance
- Nullable and unique (one-to-one) FK support

## Usage Example

```python
from sqliter import SqliterDB
from sqliter.model import BaseDBModel, ForeignKey

class Author(BaseDBModel):
    name: str
    email: str

class Book(BaseDBModel):
    title: str
    author_id: int = ForeignKey(Author, on_delete="CASCADE")

db = SqliterDB(":memory:")
db.create_table(Author)
db.create_table(Book)

author = db.insert(Author(name="John", email="john@example.com"))
book = db.insert(Book(title="My Book", author_id=author.pk))
```

## Future Work (Phase 2)

- `book.author` lazy loading access
- `author.books` reverse relationships
- Model registry for forward references

## Summary by CodeRabbit

* **New Features**
  * Foreign key support with referential integrity and configurable on-delete/on-update actions (CASCADE, SET NULL, RESTRICT, NO ACTION); automatic indexing and one‑to‑one via unique FKs; runtime FK enforcement enabled (cascade deletes supported).

* **Documentation**
  * New comprehensive Foreign Keys guide, updated model docs and site navigation entry with examples and migration notes.

* **Tests**
  * Extensive test suite covering FK configuration, enforcement, cascade semantics and edge cases.

* **Bug Fixes**
  * Improved mapping of integrity failures to clear FK-related errors and hardened insert/delete handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: docs/guide/foreign-keys.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+> The referenced table (`Author`) must be created **before** the table that
+> references it (`Book`).
+
+> [!NOTE]
+>
+> The default foreign key action is `RESTRICT` for both `on_delete` and
+> `on_update`. This means that by default, SQLiter will prevent deletion or
+> updates of referenced records if other records reference them. This is the
+> safest behavior and matches SQLite's default. You must explicitly specify
+> `on_delete="CASCADE"` or `on_update="CASCADE"` if you want cascading
```
[the remark this answers is not in the document]
Comment (seapagan) [reply docs/guide/foreign-keys.md]: not fussed with this. Github and mkdocs render these correctly
